internal/controllers/preflightvalidation_reconciler.go

```diff
@@ -0,0 +1,249 @@
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	PreflightValidationReconcilerName = "PreflightValidation"
+)
+
+// PreflightReconciler reconciles a PreflightValidation object
+type preflightValidationReconciler struct {
+	client       client.Client
+	filterAPI    *filter.Filter
+	metricsAPI   metrics.Metrics
+	helper       preflightReconcilerHelper
+	preflightAPI preflight.PreflightAPI
+}
+
+func NewPreflightValidationReconciler(
+	client client.Client,
+	filterAPI *filter.Filter,
+	metricsAPI metrics.Metrics,
+	micAPI mic.MIC,
+	kernelAPI module.KernelMapper,
+	preflightAPI preflight.PreflightAPI,
+) *preflightValidationReconciler {
+	helper := newPreflightReconcilerHelper(client, micAPI, metricsAPI, kernelAPI, preflightAPI)
+	return &preflightValidationReconciler{
+		client:       client,
+		filterAPI:    filterAPI,
+		metricsAPI:   metricsAPI,
+		helper:       helper,
+		preflightAPI: preflightAPI}
+}
+
+func (r *preflightValidationReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(PreflightValidationReconcilerName).
+		For(&v1beta2.PreflightValidation{}, builder.WithPredicates(filter.PreflightReconcilerUpdatePredicate())).
+		Owns(&kmmv1beta1.ModuleImagesConfig{}).
+		Watches(
+			&kmmv1beta1.Module{},
+			handler.EnqueueRequestsFromMapFunc(r.filterAPI.EnqueueAllPreflightValidations),
+			builder.WithPredicates(filter.PreflightReconcilerUpdatePredicate()),
+		).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: 1,
+		}).
+		Complete(
+			reconcile.AsReconciler[*v1beta2.PreflightValidation](r.client, r),
+		)
+}
+
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=modules,verbs=get;list;watch
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=preflightvalidations,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=preflightvalidations/status,verbs=get;update;patch
+
+// Reconcile Reconiliation entry point
+func (r *preflightValidationReconciler) Reconcile(ctx context.Context, pv *v1beta2.PreflightValidation) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx).WithValues("preflight name", pv.Name)
+	log.Info("Start PreflightValidation Reconciliation")
+
+	// we want the metric just to signal that this customer uses preflight
+	r.metricsAPI.SetKMMPreflightsNum(1)
+
+	modsWithMapping, modsWithoutMapping, err := r.helper.getModulesData(ctx, pv)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("getModulesData failed: %v", err)
+	}
+
+	err = r.helper.updateStatus(ctx, modsWithMapping, modsWithoutMapping, pv)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("updateStatus failed: %v", err)
+	}
+
+	err = r.helper.runPreflightValidation(ctx, modsWithMapping, pv)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("runPreflightValidation failed: %v", err)
+	}
+
+	reconCompleted := r.preflightAPI.AllModulesVerified(pv)
+	if reconCompleted {
```

Comment:
```suggestion
	if r.preflightAPI.AllModulesVerified(pv) {
```

We are not using `reconCompleted` anywhere else.

Reply:
fixed